src/Mod/PartDesign/Gui/ReferenceSelection.cpp

```diff
@@ -340,7 +340,7 @@ bool getReferencedSelection(const App::DocumentObject* thisObj, const Gui::Selec
 
 QString getRefStr(const App::DocumentObject* obj, const std::vector<std::string>& sub)
 {
-    if (obj == nullptr)
+    if (!obj)

```

Comment:
I'm just curious for some TechDraw refactoring. Is this the same as a nullptr check and why is this preferred?

Reply:
!obj means there is no object this is the same as if obj == nullptr but is seen as not the optimal coding style.

--------
For your TechDraw review, please check for unused includes. TechDraw has a lot of them. I already reviewed the ViewProviders few months back, but the more important part are the files in App and of course the .h files.

Reply:
Hum, how can I know if an include is used or not, without removing them one at a time and checking if the compiler fails? @donovaly

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-if

Reply:
> how can I know if an include is used or not, without removing them one at a time and checking if the compiler fails?

Exactly this way ;-)
But assure that you don't use precompiled headers.

I do it this way:
- search for the name of the include. Fro example for `DocumentObject`
- if I have no hits, I comment the include
- I repeat this for all includes
- from time to time I run the compiler.
- depending on the compiler output I re-activate some.

Time by time one becomes quicker, because for e.g. Exception.h one must also search for "assert".

This is a lot of work, so a project for weeks. 2 weeks ago I did this for the App part of the Part WB and it took me 2 weeks, but was also relaxing because after a stressy day just doing half an hour something repetitive is something I don't have usually during the day.